src/goto-instrument/contracts/dynamic-frames/dfcc_dsl_wrapper_program.cpp

```diff
@@ -0,0 +1,1141 @@
+  dest.destructive_append(link_deallocated_contract);
+  dest.destructive_append(postconditions);
+  dest.destructive_append(postamble);
+}
+
+void dfcc_dsl_wrapper_programt::encode_requires_write_set()
+{
+  log.debug() << "->dfcc_dsl_wrapper_programt::encode_requires_write_set()"
+              << messaget::eom;
+
+  PRECONDITION(requires_write_set_symbol_opt);
+  PRECONDITION(addr_of_requires_write_set_symbol_opt);
+
+  // call write_set_create(
+  //   requires_write_set,
+  //   assigns_size = 0,
+  //   frees_size = 0,
+  //   replacement_mode = false,
+  //   assume_requires_ctx = contract_mode == check,
+  //   assert_requires_ctx = contract_mode != check,
+  //   assume_ensures_ctx = false,
+  //   assert_ensures_ctx = false,
+  // )
+  const auto write_set = requires_write_set_symbol_opt->symbol_expr();
+  preamble.add(goto_programt::make_decl(write_set));
+
+  const auto address_of_write_set =
+    addr_of_requires_write_set_symbol_opt->symbol_expr();
+  preamble.add(goto_programt::make_decl(address_of_write_set));
+  preamble.add(goto_programt::make_assignment(
+    address_of_write_set, address_of_exprt(write_set)));
+
+  auto function_symbol =
+    library.dfcc_fun_symbol.at(dfcc_funt::WRITE_SET_CREATE).symbol_expr();
+  code_function_callt call(function_symbol);
+  auto &arguments = call.arguments();
+
+  // write set
+  arguments.emplace_back(address_of_write_set);
+
+  // max assigns clause size
+  arguments.emplace_back(from_integer(0, size_type()));
+
+  // max frees clause size
+  arguments.emplace_back(from_integer(0, size_type()));
+
+  // replacement mode
+  arguments.emplace_back((exprt)false_exprt());
+
+  if(contract_mode == dfcc_contract_modet::CHECK)
+  {
+    // assume_requires_ctx
+    arguments.emplace_back((exprt)true_exprt());
+
+    // assert_requires_ctx
+    arguments.emplace_back((exprt)false_exprt());
+  }
+  else
+  {
+    // assume_requires_ctx
+    arguments.emplace_back((exprt)false_exprt());
+
+    // assert_requires_ctx mode
+    arguments.emplace_back((exprt)true_exprt());
+  }
+
+  // assume_ensures_ctx mode
+  arguments.emplace_back((exprt)false_exprt());
+
+  // assert_ensures_ctx mode
+  arguments.emplace_back((exprt)false_exprt());
+
+  preamble.add(goto_programt::make_function_call(call));
+
+  // check for absence of allocation/deallocation in requires clauses
+  // ```c
+  // DECL __check_no_alloc: bool;
+  // CALL __check_no_alloc = check_empty_alloc_dealloc(write_set);
+  // ASSERT __check_no_alloc;
+  // DEAD __check_no_alloc: bool;
+  // ```
+  auto check_var = get_fresh_aux_symbol(
+                     bool_typet(),
+                     id2string(wrapper_symbol.name),
+                     "__no_alloc_dealloc_in_requires",
+                     source_locationt{}, // TODO add property class and comment
+                     wrapper_symbol.mode,
+                     goto_model.symbol_table)
+                     .symbol_expr();
+
+  postamble.add(goto_programt::make_decl(check_var));
```

Comment:
Applies to all `.add(...)` calls from here onwards: Missing source location.

Reply:
+n, where n is large